setup.py

```diff
@@ -1,54 +1,110 @@
-from setuptools import setup, find_packages
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
 
+"""The setup script."""
 
-PACKAGE_NAME = 'aicsimageio'
+from setuptools import find_packages, setup
 
+with open("README.md") as readme_file:
+    readme = readme_file.read()
 
-"""
-Notes:
-We get the constants MODULE_VERSION from
-See (3) in following link to read about versions from a single source
-https://packaging.python.org/guides/single-sourcing-package-version/#single-sourcing-the-version
-"""
+requirements = [
+    "numpy>=1.16",
+    "imageio>=2.3.0",
+    "scipy>=1.1.0",
+    "tifffile==0.15.1",
+]
 
-MODULE_VERSION = ""
-exec(open(PACKAGE_NAME + "/version.py").read())
+test_requirements = [
+    "codecov",
+    "flake8",
+    "pytest",
+    "pytest-cov",
+    "pytest-raises",
+]
 
-def readme():
-    with open('README.md') as f:
-        return f.read()
+dev_requirements = [
+    "bumpversion>=0.5.3",
+    "coverage>=5.0a4",
+    "flake8>=3.7.7",
+    "ipython>=7.5.0",
+    "pytest>=4.3.0",
+    "pytest-cov==2.6.1",
+    "pytest-raises>=0.10",
+    "pytest-runner>=4.4",
+    "Sphinx>=2.0.0b1",
+    "tox>=3.5.2",
+    "twine>=1.13.0",
+    "wheel>=0.33.1",
+]
 
-test_deps = ['pytest', 'pytest-cov', 'pytest-raises']
-lint_deps = ['flake8']
-interactive_dev_deps = []
-extras = {
-      'test_group': test_deps,
-      'lint_group': lint_deps,
-      'interactive_dev_group': interactive_dev_deps
+setup_requirements = [
+    "pytest-runner",
+]
+
+interactive_requirements = [
+    "altair",
+    "jupyterlab",
+    "matplotlib",
+    "pillow",
+]
+
+extra_requirements = {
+    "test": test_requirements,
+    "dev": dev_requirements,
+    "setup": setup_requirements,
+    "interactive": interactive_requirements,
+    "all": [
+        *requirements,
+        *test_requirements,
+        *setup_requirements,
+        *dev_requirements,
+        *interactive_requirements
+    ]
 }
 
-setup(name=PACKAGE_NAME,
-      version=MODULE_VERSION,
-      description='A generalized scientific image processing module from the Allen Institute for Cell Science.',
-      long_description=readme(),
-      author='AICS',
-      author_email='!AICS_SW@alleninstitute.org',
-      license='Allen Institute Software License',
-      packages=find_packages(exclude=['tests', '*.tests', '*.tests.*']),
-      entry_points={
-          "console_scripts": [
-          ]
-      },
-      install_requires=[
-            'imageio>=2.3.0',
-            'numpy>=1.16',
-            'scipy>=1.1.0',
-            'tifffile==0.15.1'
-      ],
-
-      # For test setup. This will allow JUnit XML output for Jenkins
-      setup_requires=['pytest-runner'],
-      tests_require=test_deps,
-      extras_require=extras,
-      zip_safe=False
-      )
+setup(
+    author=(
+        "Dan Toloudis, "
```

Comment:
I wonder if we can take all the names out of this and just call the author "AICS"  (in all the source code files too). Not sure what email we'd use then, but jacksonb@alleninstitute.org has a nice ring to it.

Reply:
I'd also be fine with having my personal email there too ;)

Reply:
Check most recent version of this file: Author is now "Allen Institute for Cell Science" and author email is just myself and Matt's personal.